### PR TITLE
Set `ReadTimeout` on profiling, monitoring and OCSP HTTP servers

### PR DESCRIPTION
**Pull request by neilalexander**
Reported-by: Trail of Bits <https://www.trailofbits.com>
Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -2826,6 +2826,7 @@ func (s *Server) StartProfiler() {
 		Addr:           hp,
 		Handler:        http.DefaultServeMux,
 		MaxHeaderBytes: 1 << 20,
+		ReadTimeout:    time.Second * 5,
```
Comment (wallyqs) [inline server/server.go:2829]: 5 seconds maybe too low for the profile port? Bump to 15 seconds maybe?

Comment (neilalexander) [reply server/server.go:2829]: It's only for reading the request body, which isn't really used on the profiling port. I checked that pulling long-running profiles works and it's fine.

**Comment by derekcollison** [comment]
@neilalexander let's make sure CI is green.

I think this test is failing consistently.

`TestMonitoringNoTimeout`

**Comment by neilalexander** [comment]
I've changed the monitoring port from `ReadTimeout` to `ReadHeaderTimeout` since seemingly the `TestMonitoringNoTimeout` checks `ReadTimeout` isn't set. This will mitigate the issue still.

**Comment by derekcollison** [comment]
Please make sure builds are green..

This test needs updating?

`TestProfilingNoTimeout`

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**